service/github/github_test.go

```diff
@@ -1337,9 +1336,36 @@ func TestGitHubPullRequest_Diff_fake_fallback(t *testing.T) {
 			t.Fatal(err)
 		}
 	})
+	mux.HandleFunc("/repos/o/r/compare/"+headSHA+"..."+baseSHA, func(w http.ResponseWriter, r *http.Request) {
+		if r.Method != http.MethodGet {
+			t.Errorf("unexpected access: %v %v", r.Method, r.URL)
+		}
+		if accept := r.Header.Get("Accept"); accept != "application/vnd.github.v3+json" {
+			t.Errorf("Accept header doesn't contain 'diff': %v", accept)
+		}
+
+		mergeBaseSha := "HEAD^"
+
+		commitsComparisonJSON, err := json.Marshal(github.CommitsComparison{
+			MergeBaseCommit: &github.RepositoryCommit{
+				SHA: &mergeBaseSha,
+			},
+		})
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		if _, err := w.Write(commitsComparisonJSON); err != nil {
+			t.Fatal(err)
+		}
+	})
 	ts := httptest.NewServer(mux)
 	defer ts.Close()
 
+	if err := os.Setenv("SKIP_GIT_FETCH", "true"); err != nil {
```

Comment:
Use t.SetEnv https://pkg.go.dev/testing#T.Setenv

Reply:
I use it: https://github.com/reviewdog/reviewdog/pull/1714/commits/bd6b2cdea99e80e65d3715f7b16e9b66fb48b213